### ISSUE TITLE
nnUNet: plan.json for downstream trainings

### ISSUE THREAD

**Issue by lorinczszabolcs** [Luffy03/Large-Scale-Medical#15]
Hi!

First of all, awesome work, thanks for the contribution!

I wanted to ask if you would be able to provide the `plans.json`, `dataset.json`, etc. generated configuruation files for nnUNet that were created throughout the pre-training step, as the weights by themselves are not usable straight away for the fine-tuning on downstream tasks, at the moment if I'm not mistaken we would need to do the `plan and preprocess` step on the pre-training data set, but that requires quite a lot of space and computation.

Thanks again, and looking forward to your response!

**Comment by Luffy03** [comment]
Hi, many thanks for your attention to our work!
For ssl pre-training, we also use the monai framework to train the nnunet backbone, thus the plans.json and dataset.json are not required. If you want to try fully-supervised training on VoComni, you can check the files at https://github.com/Luffy03/Large-Scale-Medical/tree/main/VoComni/nnUNet_preprocessed. Hope they can help.

**Comment by lorinczszabolcs** [comment]
Thanks for the quick answer!

I'm mainly interested in using the SSL-pre-trained weights for fine-tuning on downstream tasks (using the nnUNet codebase). For that the plans and dataset.json are required if we want to initialize the model instance, for which we can load the shared weights. Or is there a class in the repository that has some hard-coded architecture implemented already that we could use for loading the pre-trained weights?

**Comment by Luffy03** [comment]
Oh, you mean the downstream. Please refer to [Downstream/nnUNet](https://github.com/Luffy03/Large-Scale-Medical/tree/main/Downstream/nnUNet), where I provide the usage instructions. Detailed codes at [nnUNetTrainer_pretrain.py](https://github.com/Luffy03/Large-Scale-Medical/blob/main/Downstream/nnUNet/nnunetv2/training/nnUNetTrainer/nnUNetTrainer_pretrain.py). 
You only need to use the original preprocessing implementations then employ "nnUNetv2_train xxx 3d_fullres all -tr **nnUNetTrainer_pre**" instead of "nnUNetv2_train xxx 3d_fullres all -tr nnUNetTrainer".
```
cd nnUNet
source activate YOUR-CONDA-ENVIRONMENT
nnUNetv2_plan_and_preprocess -d xxx -c 3d_fullres --verbose --verify_dataset_integrity
nnUNetv2_train xxx 3d_fullres all -tr nnUNetTrainer_pre
```

**Comment by lorinczszabolcs** [comment]
Thank you for getting back!

I saw the instructions there, but I was wondering if we could skip the `nnUNetv2_plan_and_preprocess` step, as in the original nnUNet repository `ExperimentPlanner` creates a new architecture that is specific to each data set that is being used.

I now checked and realized that the architecture is fixed [here](https://github.com/Luffy03/Large-Scale-Medical/blob/ab6829393c6a7e7623213406abc7340e0b8f45a8/Downstream/nnUNet/nnunetv2/experiment_planning/experiment_planners/default_experiment_planner.py#L345-L369) in this repository, so I assume I can just run `plan_and_preprocess` on my downstream task data set and then load the weights that were shared, because the architecture won't differ due to this :). But please feel free to correct me if I'm wrong.

The only downside is that it is not compatible with original nnUNet implementation anymore with the `ExperimentPlanner` overrided.

**Comment by Luffy03** [comment]
Hi, as I know the default Experimentplanner will not change the architecture of PlainConvUNet, so you can still inherit the pre-trained weights. The only difference is the pre-process strategies. You kindly mentioned that "ExperimentPlanner creates a new architecture that is specific to each data set", is this a recent update?

**Comment by Luffy03** [comment]
Yes, some parameters like patch_size will change but the network parameters will not change. You can try our codes and you will find that most of the network parameters are loaded. The data processing will be different, we are still exploring this part.

**Comment by lorinczszabolcs** [comment]
Ok, thanks! Will get back once I was able to try it out.

**Comment by Luffy03** [comment]
Hi, FYI, the codes https://github.com/Luffy03/Large-Scale-Medical/blob/ab6829393c6a7e7623213406abc7340e0b8f45a8/Downstream/nnUNet/nnunetv2/training/nnUNetTrainer/nnUNetTrainer_pretrain.py#L135 will print the loaded network paremeters.
